scripts/GTF_OVERLAP_annotation.Rmd

```diff
@@ -0,0 +1,640 @@
+  dataframe %>%
+    mutate(!!sym(feature_name) := ifelse(grepl(feature_name, attribute), 
+        # if TRUE, replace entire value by regex
+        str_replace(string=attribute, pattern='.*', 
+                    replacement=stri_extract_first_regex(
+                      attribute, paste0('(?<=.{0,100}\\b',
+                                        feature_name,'\\b\\s{0,100})[\\w\\d-]+[)]*')
+                    )), # if this does not work, remove teh second {0,100}
+        # if FALSE, replace by NA
+        NA)
+           )
+}
+
+# loop over the list of feature names and extract each feature from the attributes column
+for (feature in variable_names) {
+  df_combined <- extract_feature(df_combined, feature)
+  print(paste0("done adding ",feature))
+}
+
+# saving the table just in case
+file_path <- paste(save_path, 'gene_annotations.csv', sep='')
+write_csv(df_combined, file_path)
+```
+
+# Raw statistics
+
+Including total peak counts (boxplots), feature distribution (exons, CDS, gene, transcript, non-defined).
+
+```{r}
+# import annotated table
+df_combined = read.csv(paste(save_path, "gene_annotations.csv", sep=''))
+
+# make a copy of df_combined just in case
+df_copy = copy(df_combined)
+```
+
+## Total peak count (Boxplot)
+
+All distinct peaks in general, or those only present in genes (you can choose for yourself which one you would prefer).
+
+```{r fig.height=4, fig.width=6}
+# peak counts df
+peak_counts <- df_combined %>%
+  group_by(sample) %>%
+  summarise(count = n_distinct(peak_id))
+peak_counts$condition <- c(rep("gof",3),rep("wt",3))
+
+# sometimes it's easier to just manually write down the hits from wc -l from the pureclip bed files
+peak_counts <- data.frame(
+  sample=c("gof_1" ,"gof_2", "gof_3", "wt_1", "wt_2", "wt_3"),
+  count=c(3392, 3040, 2647, 1446, 3011, 2509),
+  condition=c(rep("gof",3),rep("wt",3))
+)
+
+ggplot(peak_counts, aes(x=as.factor(condition), y=count)) +
+    geom_boxplot(fill="slateblue2", alpha=0.2) +
+    geom_text_repel(data=peak_counts,aes(label=sample)
+                    # , position = position_jitter(seed = 1)
+                    ) +
+    geom_point(color="blue", size=2, alpha=0.9,
+                # position = position_jitter(seed = 1)
+               ) +
+  
+    # coord_cartesian(ylim = c(0, 3500)) + # define range
+    xlab("condition") +
+    ylab("peak count")
+
+  # save figure
+ggsave(paste(save_path, "boxplots/pureclip_raw_peak_count.png", sep=''))
+```
+
+## Feature distribution (Piechart)
+
+Distribution of features that intersected with the peak bed file. The goal is to ensure 1 peak is associated with 1 feature and not several features. The priority order we enforce is CDS \> 5' UTR \> 3' UTR \> intron. This order was already incorporated in the .sh by merging the peak bed files in this exact order.
+
+```{r echo=FALSE, fig.align="center", fig.height=5, fig.width=9, message=FALSE, warning=FALSE}
+p <- '/exports/sascstudent/amhsilie/iclip_downstream/gtfs/'
```

Comment:
Is this file meant to be used by the user of the pipeline? In that case this path should probably not be hard-coded.